UnitTests/Tests.cs

```diff
@@ -1226,13 +1212,13 @@ public void SafeReadCanBeUsedInParallel()
             {
                 for (var i = 0; i < 5; i++)
                 {
-                    var position = this.random.Next(0, bufferLength);
-                    var startPosition = position;
-                    var length = this.random.Next(0, bufferLength - position);
+                    long position = this.random.Next(0, bufferLength);
+                    long startPosition = position;
+                    var length = this.random.Next(0, (int)(bufferLength - position));
                     var readBuffer = new byte[length];
```

Comment:
Why the change? `var` assumes the correct type.

You can keep `var` if you use the correct literal: `0L`.